marketplace/netlify-build/src/index.js

```diff
@@ -0,0 +1,81 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import PropTypes from 'prop-types';
+import { init } from 'contentful-ui-extensions-sdk';
+import { createPubSub } from './pubnub-client';
+import { Select, Option } from '@contentful/forma-36-react-components';
+import NeflifySideBarBuildButton from './NetlifySideBarBuildButton';
+
+function createSiteStructure(parameters = {}) {
+  const buildHooks = (parameters.buildHookUrls || '').split(',');
+  const channels = (parameters.channels || '').split(',');
+  const siteIds = (parameters.netlifySiteIds || '').split(',');
+  const names = (parameters.names || '').split(',');
+  const urls = (parameters.netlifySiteUrls || '').split(',');
+
+  return buildHooks.reduce(
+    (acc, _, i) =>
+      acc.concat([
+        {
+          buildHookUrl: buildHooks[i],
+          channel: channels[i],
+          netlifySiteId: siteIds[i],
+          name: names[i],
+          url: urls[i]
+        }
+      ]),
+    []
+  );
+}
+
+export default class NetlifyExtension extends React.Component {
+  static propTypes = {
+    sdk: PropTypes.object.isRequired,
+    createPubSub: PropTypes.func.isRequired
+  };
+
+  state = {
+    users: [],
+    sites: createSiteStructure(this.props.sdk.parameters.installation),
+    selectedSiteIndex: 0
+  };
+
+  async componentDidMount() {
+    this.props.sdk.window.startAutoResizer();
+    const { items } = await this.props.sdk.space.getUsers();
+    this.setState({ users: items });
+  }
+
+  selectSite = e => {
+    this.setState({ selectedSiteIndex: Number(e.target.value) });
```

Comment:
I wouldn't use `new`-less constructors, just do `parseInt(value, 10)`

Reply:
Any specific reason for that?

Reply:
It just confuses me when I read the code - I'm not sure what the behavior is, I'm not aware of edge cases. Also with `parseInt` I see I'm dealing with integers. For different class of errors I'd expect constructors to be called with `new` and here we just use it as a function.

Reply:
Ok thanks for the explanation. In this specific use case the functionality is identical. However, I think readability will trump semantics ;) I'll change it.